org.eclipse.Java.json

```diff
@@ -0,0 +1,52 @@
+{
+  "id" : "org.eclipse.Java",
+  "branch" : "4.13",
+  "runtime" : "org.gnome.Sdk",
+  "runtime-version" : "3.34",
+  "sdk" : "org.gnome.Sdk",
+  "sdk-extensions" : [ "org.freedesktop.Sdk.Extension.openjdk11" ],
+  "command" : "eclipse",
+  "modules" : [ {
+    "name" : "openjdk",
+    "buildsystem" : "simple",
+    "build-commands" : [ "/usr/lib/sdk/openjdk11/installjdk.sh" ]
+  }, {
+    "name" : "eclipse",
+    "buildsystem" : "simple",
+    "build-commands" : [
+      "mv eclipse /app",
+      "mkdir -p /app/bin",
+      "ln -s /app/eclipse/eclipse /app/bin",
+      "install -Dm 644 org.eclipse.Java.desktop /app/share/applications/org.eclipse.Java.desktop",
+      "install -Dm 644 org.eclipse.Java.appdata.xml /app/share/metainfo/org.eclipse.Java.appdata.xml",
+      "mkdir -p /app/share/icons/hicolor/32x32/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse32.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse32.png /app/share/icons/hicolor/32x32/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/64x64/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse64.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse64.png /app/share/icons/hicolor/64x64/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/128x128/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse128.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse128.png /app/share/icons/hicolor/128x128/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/256x256/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse256.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse256.png /app/share/icons/hicolor/256x256/apps/org.eclipse.Java.png ; fi",
+      "sed -i -e '/osgi.configuration.area/d' /app/eclipse/eclipse.ini",
+      "echo \"-Dosgi.configuration.area=@user.home/.var/app/org.eclipse.Java/eclipse/configuration\" >> /app/eclipse/eclipse.ini",
+      "echo \"--patch-module=java.base=/app/eclipse/flatpak-dev-shim.jar\" >> /app/eclipse/eclipse.ini"
+    ],
+    "sources" : [ {
+      "type" : "archive",
+      "url" : "https://download.eclipse.org/technology/epp/downloads/release/2019-09/R/eclipse-java-2019-09-R-linux-gtk-x86_64.tar.gz",
+      "sha512" : "eb408902f079d6666863bc318a0586589be9a86e4cd57125ef1f97eb4f4a9d6b70aa52ea23129f5f95eb513c3ce1889683516d91e85a484fcae7328fa8e1eeff",
+      "strip-components" : 0
+    }, {
+      "type" : "file",
+      "path" : "org.eclipse.Java.desktop",
+      "dest-filename" : "org.eclipse.Java.desktop"
+    }, {
+      "type" : "file",
+      "path" : "org.eclipse.Java.appdata.xml",
```

Comment:
New metainfo files should use .metainfo instead of .appdata and install in $datadir/metainfo.

Reply:
Fixed in 6452877